staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -257,11 +257,11 @@ type ObjectMeta struct {
 	// +patchStrategy=merge
 	Finalizers []string `json:"finalizers,omitempty" patchStrategy:"merge" protobuf:"bytes,14,rep,name=finalizers"`
 
-	// The name of the cluster which the object belongs to.
-	// This is used to distinguish resources with same name and namespace in different clusters.
-	// This field is not set anywhere right now and apiserver is going to ignore it if set in create or update request.
-	// +optional
-	ClusterName string `json:"clusterName,omitempty" protobuf:"bytes,15,opt,name=clusterName"`
+	// ClusterName tombstone. It consumed proto tag 15. We will remove
+	// completely sans a comment here next release (1.25). The name is
+	// changed here for a release to ensure any users (which we do not
+	// expect) will get a compile error before a serialization error.
```

Comment:
keep the `+optional` comment, and add `Deprecated: ...` to trigger govet warnings on use as well

Reply:
done